interop/shared/src/main/scala/scalaz/zio/interop/IONewtype.scala

```diff
@@ -0,0 +1,17 @@
+package scalaz.zio
+package interop
+
+private[interop] trait IONewtype {
+  type Base
+  trait Tag extends Any
+
+  type T[+E, +A] <: Base with Tag
+
+  def apply[E, A](io: IO[E, A]): T[E, A] =
+    io.asInstanceOf[T[E, A]]
+
+  def unwrap[E, A](t: T[E, A]): IO[E, A] =
+    t.asInstanceOf[IO[E, A]]
+}
+
+private[interop] object Par extends IONewtype
```

Comment:
What does this newtype gain us that `@@` already didn't give us? It seems like this is just to get `Par` and `Par.T`, but one could easily just 

```scala
type ParIO[E, A] = IO[E, A] @@ Parallel
```
For some static `Parallel`. You've tied to `Par` to `IO`, instead of `F[_, _]`, so there's not much abstraction going on that would justify the code duplication.

Reply:
This is not an abstraction. It's a workaround, as `@@` wouldn't work if we don't have Scalaz in the classpath. A simple setup could show the problem:

build.sbt
```
scalaVersion := "2.12.7"

libraryDependencies += "org.scalaz" %% "scalaz-zio-interop" % "using-@@-instead-of-newtype"

libraryDependencies += "org.typelevel" %% "cats-effect" % "1.0.0"
```
in console
```
scala> import scalaz.zio.interop._
import scalaz.zio.interop._

scala> import catz._
import catz._

scala> type ParTask[A] = scalaz72.ParIO[Throwable, A]
<console>:17: error: Symbol 'type scalaz.package.$at$at' is missing from the classpath.
This symbol is required by 'type scalaz.zio.interop.scalaz72.ParIO'.
Make sure that type @@ is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
A full rebuild may help if 'scalaz72.class' was compiled against an incompatible version of scalaz.package.
       type ParTask[A] = scalaz72.ParIO[Throwable, A]
```